src/NServiceBus.Core/Transports/Msmq/ReceiveWithTransactionScope.cs

```diff
@@ -22,6 +22,11 @@ public ReceiveWithTransactionScope(TransactionOptions transactionOptions)
             {
                 var message = inputQueue.Receive(TimeSpan.FromMilliseconds(10), MessageQueueTransactionType.Automatic);
 
+                if (message == null)
```

Comment:
I thought we discussed the null check is not necessary. At least you tried to convince me about that. If that is still true why was id added here?